shogun-lib/src/main/java/de/terrestris/shogun/lib/service/ImageFileService.java

```diff
@@ -89,4 +95,63 @@ public void isValidType(String contentType) throws InvalidContentTypeException {
         }
     }
 
+    @Override
+    public ImageFile create(MultipartFile uploadFile, Boolean writeToSystem) throws Exception {
+        if (!writeToSystem) {
+            return this.create(uploadFile);
+        }
+
+        String uploadBasePath = uploadProperties.getPath();
+        if(uploadBasePath == null || uploadBasePath.isEmpty()) {
+            throw new Exception("Could not upload file. uploadBasePath is null.");
+        }
+        String fileName = uploadFile.getOriginalFilename();
+        if(fileName == null || fileName.isEmpty()) {
+            throw new Exception("Could not upload file. fileName is null.");
+        }
+
+        FileUtil.validateFile(uploadFile);
```

Comment:
Can we move the validation to the `isValidType` method which is running on start?